### PR TITLE
Shard tensor skill

### PR DESCRIPTION
**Pull request by coreyjadams**
# PhysicsNeMo Pull Request

This PR adds a skill for physicsnemo shard tensor integration.

Main areas this skill helps with (AI Generated Skill summary ....)
- Retrofitting scripts for domain parallelism — converting single-GPU or DDP training/inference scripts to ShardTensor (ddp × domain mesh, scatter_tensor, batch-size-1 rule) without touching model code
- Picking the right wrapper — DDP vs FSDP2 decision table, domain-axis weight broadcast, spatial-param sharding, and why FSDP1 is never the answer
- Enabling unsupported ops — when a MissingShardPatch or wrong numerics warrants a shard patch, and how to register one from user code without forking physicsnemo
- Debugging the sharp edges — the swallowed-TypeError trap, silent requires_grad_ no-op, unwaited collectives, and torch.compile constraints (regional compile, dynamic=False)
- Verifying correctness — bootstrapping multi-GPU tests with numerical_shard_tensor_check, always comparing gradients (not just forward) against a single-GPU run

Measured impact: agents with the skill found the right approach with ~half the tokens (+44% efficiency), routed to it reliably (+48% discoverability), and avoided the FSDP1/model-editing mistakes baselines made (+16% correctness).

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: skills/physicsnemo-shard-tensor/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        if not isinstance(p, DTensor):
+            torch.distributed.broadcast(p.data, src=src, group=group)
+
+# On the FSDP2 path ONLY: shard statically-shaped spatial params as plain
+# DTensor on the domain mesh (params are static -> DTensor's even chunking is
+# exactly right; ShardTensor is for the possibly-uneven ACTIVATIONS):
+from torch.distributed.tensor import distribute_tensor
+model.pos_embed = nn.Parameter(
+    distribute_tensor(model.pos_embed.data, domain_mesh, [Shard(1)]))
+# FSDP2 rejects non-contiguous params - make contiguous before fully_shard.
```
[the remark this answers is not in the document]
Comment (coreyjadams) [reply skills/physicsnemo-shard-tensor/SKILL.md:126]: The example is updated in #1682

**Review by melo-gonzo — APPROVED** [review]

**Comment by melo-gonzo** [comment]
/ok to test [ae3ced7](https://github.com/NVIDIA/physicsnemo/pull/1892/commits/ae3ced71671af44c643ff7dfdb30cd9ba338f326)
